internal/server/rbac/roles.go

```diff
@@ -742,55 +755,46 @@ func claimListToMap(claims []rbacapi.Claim) map[string][]string {
 	return claimMap
 }
 
-func replaceClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) {
-	// Step through the existing annotations and for each that looks like a claim
-	// annotation, remove it if the corresponding claim is not in the new claims
-	// map.
-	for k := range sa.Annotations {
-		if name, ok := rbacapi.OIDCClaimNameFromAnnotationKey(k); ok {
-			if _, exists := claims[name]; !exists {
-				delete(sa.Annotations, k)
-			}
-		}
-	}
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
-	}
-	// Add or update the annotations for the new claims
-	for name, values := range claims {
-		sa.Annotations[rbacapi.AnnotationKeyOIDCClaim(name)] = strings.Join(values, ",")
-	}
+func replaceClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) error {
+	return rbacapi.SetOIDCClaimsAnnotation(sa, claims)
 }
 
-func amendClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) {
+func amendClaimAnnotations(sa *corev1.ServiceAccount, claims map[string][]string) error {
+	existingClaims, err := rbacapi.OIDCClaimsFromAnnotationValues(sa.Annotations)
+	if err != nil {
+		return fmt.Errorf("failed to parse OIDC claims from annotation values: %w", err)
+	}
 	for name, values := range claims {
-		if existing, exists := sa.Annotations[rbacapi.AnnotationKeyOIDCClaim(name)]; exists {
-			values = append(strings.Split(existing, ","), values...)
-		}
-		slices.Sort(values)
-		values = slices.Compact(values)
-		if sa.Annotations == nil {
-			sa.Annotations = map[string]string{}
+		if _, exists := existingClaims[rbacapi.AnnotationKeyOIDCClaim(name)]; exists {
```

Comment:
Calling `rbacapi.AnnotationKeyOIDCClaim()` seems its going to give you back an old style annotation key, but the map of existing claims have _just_ the claim name as the keys.

```suggestion
		if _, exists := existingClaims[name]; exists {
```